mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/query/parsing/nativequery/impl/MongoDBQueryDescriptorBuilder.java

```diff
@@ -146,19 +145,74 @@ public boolean setCollation(String collation) {
 	}
 
 	public MongoDBQueryDescriptor build() {
-		if ( operation == Operation.DISTINCT ) {
-			return new MongoDBQueryDescriptor( collection, operation, parse( criteria ), parseCollation( collation ), distinctFieldName );
-		}
-		else if ( operation != Operation.AGGREGATE_PIPELINE ) {
-			return new MongoDBQueryDescriptor(
-				collection,
-				operation,
-				parse( criteria ),
-				parse( projection ),
-				parse( orderBy ),
-				parse( options ),
-				parse( updateOrInsert ),
-				null );
+		//@todo redactor the spagetti!
```

Comment:
Make sur eto refactor only if you need to. Otherwise, only remove the comment for now